addons/l10n_ch/models/mail_template.py

```diff
@@ -29,26 +29,21 @@ def generate_email(self, res_ids, fields):
             for record in self.env[self.model].browse(res_ids):
                 inv_print_name = self._render_field('report_name', record.ids, compute_lang=True)[record.id]
                 new_attachments = []
-
-                if record.l10n_ch_isr_valid:
+                record_dict = multi_mode and result[record.id] or result
+                if record.l10n_ch_is_qr_valid:
+                    # We add an attachment containing the QR-bill
+                    qr_report_name = 'QR-' + inv_print_name + '.pdf'
+                    qr_pdf = self.env.ref('account.account_invoices')._render_qweb_pdf(record.ids)[0]
+                    qr_pdf = base64.b64encode(qr_pdf)
+                    new_attachments.append((qr_report_name, qr_pdf))
+                elif record.l10n_ch_isr_valid:
                     # We add an attachment containing the ISR
                     isr_report_name = 'ISR-' + inv_print_name + '.pdf'
                     isr_pdf = self.env.ref('l10n_ch.l10n_ch_isr_report')._render_qweb_pdf(record.ids)[0]
                     isr_pdf = base64.b64encode(isr_pdf)
                     new_attachments.append((isr_report_name, isr_pdf))
-
-                if record.partner_bank_id._eligible_for_qr_code('ch_qr', record.partner_id, record.currency_id):
-                    # We add an attachment containing the QR-bill
-                    qr_report_name = 'QR-bill-' + inv_print_name + '.pdf'
-                    qr_pdf = self.env.ref('l10n_ch.l10n_ch_qr_report')._render_qweb_pdf(record.ids)[0]
-                    qr_pdf = base64.b64encode(qr_pdf)
-                    new_attachments.append((qr_report_name, qr_pdf))
-
-                record_dict = multi_mode and result[record.id] or result
-                attachments_list = record_dict.get('attachments', False)
-                if attachments_list:
-                    attachments_list.extend(new_attachments)
                 else:
-                    record_dict['attachments'] =  new_attachments
-
+                    #If neither qr nor isr is printable, we stick to the simple invoice.
+                    new_attachments = record_dict.get('attachments', False)
+                record_dict['attachments'] = new_attachments
```

Comment:
Is this whole override still needed since the QR/ISR is automatically append to the invoice PDF, itself embedded into the mail compose wizard ?